### PR TITLE
boards: sam_v71_xult: add support for CAN-FD

### PR DESCRIPTION
**Pull request by iperry**
CAN1 is connected to an onboard ATA6561 transceiver on the SAM V71
Xplained Ultra board. This changeset adds the required DTS plumbing.

Tested with:
  - samples/drivers/can

### CODE REVIEW

**Review by mbolivar-nordic — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
Please add can + canfd capabilities to the board yaml file as well as documentation.

File: boards/arm/sam_v71_xult/sam_v71_xult_defconfig [hunk 1]
```diff
@@ -14,6 +14,7 @@ CONFIG_USART_SAM=y
 CONFIG_BOARD_SAM_V71_XULT=y
 CONFIG_WDT_DISABLE_AT_BOOT=y
 CONFIG_BUILD_OUTPUT_HEX=y
+CONFIG_GPIO=y
```
Comment (nandojve) [inline boards/arm/sam_v71_xult/sam_v71_xult_defconfig:17]: Why is this necessary?

**Review by nandojve — CHANGES_REQUESTED** [review]

File: boards/arm/sam_v71_xult/sam_v71_xult_defconfig (hunk 1 above)
Comment (mbolivar-nordic) [reply boards/arm/sam_v71_xult/sam_v71_xult_defconfig:17]: I was discussing with @iperry offline. Most boards have CONFIG_GPIO=y, so samples like samples/drivers/can don't even bother to set CONFIG_GPIO=y, even though blinky does. This is causing a twister error.

And the board porting guidelines (https://docs.zephyrproject.org/latest/guides/porting/board_porting.html#general-recommendations) do say that upstream boards should enable any drivers needed for LEDs, buttons, etc., which this board doesn't seem to be doing. So it seemed OK to add it here.

**Comment by mbolivar-nordic** [comment]
> nandojve added this to the [v3.0.1](/zephyrproject-rtos/zephyr/milestone/58) milestone

Did you really mean 3.0.1, or was this supposed to be 3.1.0? (I wasn't aware we were planning a 3.0.1 release is why I ask -- not saying it's not going to happen, just curious)

File: boards/arm/sam_v71_xult/sam_v71_xult_defconfig (hunk 1 above)
Comment (nandojve) [reply boards/arm/sam_v71_xult/sam_v71_xult_defconfig:17]: > samples/drivers/can fails to compile because the CAN example blinks an LED.

This means that CAN should select GPIO at prj.conf instead at board level. This is a sample config error and not a missing board configuration. At board level only configs that are strictly necessary to put board to run should be enabled, otherwise firmware can be bigger than expected.

**Review by henrikbrixandersen — COMMENTED** [review]
You could squash the three commits touching the board files, but it's not a blocker.

Please be patient as we are in feature freeze for v3.0.0, which should be released within the next week or so.

**Review by henrikbrixandersen — APPROVED** [review]

**Review by mbolivar-nordic — APPROVED** [review]

File: boards/arm/sam_v71_xult/sam_v71_xult_defconfig (hunk 1 above)
Comment (mbolivar-nordic) [reply boards/arm/sam_v71_xult/sam_v71_xult_defconfig]: @nandojve you've marked this as resolved but you still have a -1 -- is there something else you want to see?

Comment (iperry) [reply boards/arm/sam_v71_xult/sam_v71_xult_defconfig]: I removed the first commit, squashed the remaining three, and rebased against main.

Thanks for the patience!

**Review by henrikbrixandersen — APPROVED** [review]

**Comment by iperry** [comment]
I removed the first commit, squashed the remaining three, and rebased against main.

Thanks for the patience!

**Review by nandojve — APPROVED** [review]
> Thanks for the patience!

Thank you for your contribution!

**Review by alexanderwachter — APPROVED** [review]
Thanks!
